src/impy/common.py

```diff
@@ -428,6 +429,85 @@ def __ne__(self, other_instance):
         return False
 
 
+@dataclass
+class RMMARDState:
+    _c_number: np.ndarray = None
+    _u_array: np.ndarray = None
+    _u_i: np.ndarray = None
+    _u_j: np.ndarray = None
+    _seed: np.ndarray = None
+    _counter: np.ndarray = None
+    _big_counter: np.ndarray = None
+    _sequence_number: np.ndarray = None
+
+    def _record_state(self, generator):
+        data = generator.crranma4
+
+        self._c_number = data.c
+        self._u_array = data.u
+        self._u_i = data.i97
+        self._u_j = data.j97
+        self._seed = data.ijkl
+        self._counter = data.ntot
+        self._big_counter = data.ntot2
+        self._sequence_number = data.jseq
+        return self
+
+    def _restore_state(self, generator):
+        data = generator.crranma4
+
+        data.c = self._c_number
+        data.u = self._u_array
+        data.i97 = self._u_i
+        data.j97 = self._u_j
+        data.ijkl = self._seed
+        data.ntot = self._counter
+        data.ntot2 = self._big_counter
+        data.jseq = self._sequence_number
+        return self
+
+    def __eq__(self, other: object) -> bool:
+        if other.__class__ is not self.__class__:
+            return NotImplemented
+
+        return (
+            np.array_equal(self._c_number, other._c_number)
+            and np.array_equal(self._u_array, other._u_array)
+            and np.array_equal(self._u_i, other._u_i)
+            and np.array_equal(self._u_j, other._u_j)
+            and np.array_equal(self._seed, other._seed)
+            and np.array_equal(self._counter, other._counter)
+            and np.array_equal(self._big_counter, other._big_counter)
+            and np.array_equal(self._sequence_number, other._sequence_number)
+        )
+
+    def copy(self):
+        copies = []
+        for k in self.__dataclass_fields__:
+            obj = getattr(self, k)
+            copies.append(obj.copy() if hasattr(obj, "copy") else obj)
+
+        return RMMARDState(*copies)
+
+    def __getstate__(self):
+        return self.__dict__
+
+    def __setstate__(self, state):
+        self.__dict__ = state
```

Comment:
These implementations can be removed. Class instances are automatically pickleable. What you wrote here is the default implementation of getstate/setstate for instances.

https://docs.python.org/3/library/pickle.html#what-can-be-pickled-and-unpickled

Reply:
Removed.